rpcserver.go

```diff
@@ -3314,12 +3315,56 @@ func (r *rpcServer) BurnAsset(ctx context.Context,
 		}
 	}
 
+	// At this point everything completed correctly, so we log this burn in
+	// our db.
+	err = r.cfg.AssetStore.InsertBurn(
```

Comment:
I still think this is problematic. We make a call to `RequestShipment` above. What stops us from marking the shipment as a burn? We would also inherit the logic that only marks something as a finalized transfer once fully confirmed. `InsertBurn` would also not need to read the transfer at all, since it would be run in the same db txn that the transfer is inserted within.